pkg/build/registry/build/strategy.go

```diff
@@ -60,9 +61,10 @@ func (strategy) AllowUnconditionalUpdate() bool {
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (strategy) PrepareForCreate(obj runtime.Object) {
 	build := obj.(*api.Build)
-	if len(build.Status.Phase) == 0 {
-		build.Status.Phase = api.BuildPhaseNew
-	}
+	build.Status.Duration = 0
+	build.Status.Phase = api.BuildPhaseNew
+	build.Status.StartTimestamp = nil
+	build.Status.CompletionTimestamp = nil
```

Comment:
Doesn't matter - this hook should always normalize.

fwiw, here are the defaults for builds: https://github.com/openshift/origin/blob/de5b801e8d196384bae034405f28b7b6771bf6f0/pkg/build/api/v1/conversion.go#L214-L244